cli/src/pcluster/config/cluster_config.py

```diff
@@ -1483,10 +1504,120 @@ def _register_validators(self):
         )
 
 
-class ByosQueue(SlurmQueue):
+class ByosComputeResource(SlurmComputeResource):
+    """Represent the Byos Compute Resource."""
+
+    def __init__(
+        self,
+        instance_types: List[str],
+        custom_settings: Dict = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.instance_types = instance_types
+        self.custom_settings = custom_settings
+
+
+class ByosQueue(_CommonQueue):
     """Represent the Byos queue."""
 
-    pass
+    def __init__(
+        self,
+        compute_resources: List[ByosComputeResource],
+        networking: ByosQueueNetworking,
+        custom_settings: Dict = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.compute_resources = compute_resources
+        self.networking = networking
+        self.custom_settings = custom_settings
+
+    def _register_validators(self):
```

Comment:
why not moving common validators under _CommonQueue?

Reply:
because in _CommonQueue, we didn't define the ComputeResource which is defined in children classes